<?xml version="1.0"?>
<launch>
  <!-- include file="$(find cob_3d_mapping_common)/ros/launch/transform_shape_array.launch">
    <arg name="input" value="/segmentation/shape_array"/>
    <arg name="output" value="/transformed/shape_array"/>
  </include-->

  <arg name="shape_array_in" default="/transformed/shape_array"/>
  <arg name="target_frame" default="/map"/>
  <arg name="debug_mode" default="false"/>
  <arg name="prefix" value="gdb -ex run --args" if="$(arg debug_mode)"/>
  <arg name="prefix" value="" unless="$(arg debug_mode)"/>
  <node ns="geometry_map" pkg="cob_3d_mapping_geometry_map"
        type="geometry_map_node" name="geometry_map" output="screen" launch-prefix="$(arg prefix)">
    <remap from="shape_array_in" to="$(arg shape_array_in)"/>
    <param name="file_path" type="string" value="/Bag_Files/kitchen_table_chair"/>
    <param name="save_to_file" type="bool" value="false"/>
    <param name="map_frame_id" type="string" value="$(arg target_frame)" />
    <param name="d" type="double" value="0.10" />
    <param name="cos_angle" type="double" value="0.93" /><!-- arccos(x)*180/PI -->
    <param name="enable_tf" type="bool" value ="true" />
    <param name="enable_cyl" type="bool" value="true" />
    <param name="enable_poly" type="bool" value="true" />
  </node>
</launch>
